src/ndsparse.jl

```diff
@@ -189,7 +189,7 @@ end
 
 function (==)(a::NDSparse, b::NDSparse)
     flush!(a); flush!(b)
-    return a.index == b.index && a.data == b.data
+    return isequal(a.index, b.index) && isequal(a.data, b.data)
```

Comment:
FWIW, in DataFrames `==` returns `missing` in the presence of missing values, to be consistent with arrays. Maybe you want the same thing here? It would be nice to be consistent over the language and ecosystem.

Reply:
I did not know that about DataFrames.  Do you remember the reasoning behind the decision?  I can't think of a use case where 3-valued logic would be desired.  I imagine running into `missing` via `table1 == table1` would occur by accident more often than not.  The behavior here would make it consistent with past behavior with DataValues, where `NA == NA`

Reply:
The reasoning is the same as for `[1, missing] == [1, 2]`: if there's a missing value you don't know whether the arrays are equal or not. Precisely, if a `missing` value happens by accident, it must absolutely not go unnoticed. DataValues have followed the design of C#, but Julia uses a different approach.

Reply:
My first thought was that the notion of table equality wouldn't be the same as for arrays, but I can be on board with returning `missing`